Export TypeScript types for Hydrogen Routes (#1608)
Shopify/hydrogen: docs/framework/routes.md

{% endcodeblock %}

> The `handle` property is available from the hook `useRouteParams()`:

Reviewer: ```suggestion
You can retrieve the `handle` property by using the [`useRouteParams` hook](https://shopify.dev/api/hydrogen/hooks/framework/userouteparams):
```

### Server props

In addition to `request` and `response` props, any props you manage with [`setServerProps`](https://shopify.dev/custom-storefronts/hydrogen/framework/server-props) are passed to each of your server components as props:

{% codeblock file %}

```jsx
export default function MyPage({custom, props, here}) {
  // Use custom server props
}
```

{% endcodeblock %}

## TypeScript

Hydrogen supports TypeScript out of the box. When building route components, you can use the provided TypeScript types to improve your developer experience:

{% codeblock file %}

```tsx
import type {
  HydrogenApiRoute,
  HydrogenApiRouteOptions,
  HydrogenRequest,
  HydrogenRouteProps,
} from '@shopify/hydrogen';

export default function MyPage(props: HydrogenRouteProps) {
  //
}

export const api: HydrogenApiRoute = async(request, options) => {
  //
}

// Alternate version of `api`:
export async function api(request: HydrogenRequest, options: HydrogenApiRouteOptions) {

}
```

{% endcodeblock %}

> 

Reviewer: ```suggestion

> Tip:
> The Hello World template is available in [TypeScript](https://shopify.dev/custom-storefronts/hydrogen/templates#hello-world-template). You can also refer to the [example implementation of TypeScript](https://github.com/Shopify/hydrogen/tree/main/examples/typescript) in GitHub.
```
